app/controllers/github_commits_controller.rb

```diff
@@ -2,15 +2,47 @@ class GithubCommitsController < ApplicationController
   
   unloadable
    
-  skip_before_filter :check_if_login_required
-  skip_before_filter :verify_authenticity_token
+  skip_before_action :check_if_login_required
+  skip_before_action :verify_authenticity_token
   
   before_action :verify_signature?
   
   GITHUB_URL = "https://github.com/"
   REDMINE_JOURNALIZED_TYPE = "Issue"
   REDMINE_ISSUE_NUMBER_PREFIX = "#rm"
 
+  #added JPBD Jan2022
+  def github_change_notification
+    resp_json = nil
+    # if payload contains pr and action is opened (new pr)
+    if params[:pull_request].present? && params[:pull_request][:state] == "open"
+      #get issue ID
+      pr_title = params[:pull_request][:title]
+      issue_id = pr_title.partition(REDMINE_ISSUE_NUMBER_PREFIX).last.split(" ").first.to_i
+      issue = Issue.find_by(id: issue_id)
+      #if issue id exists in redmine & status is in progress (2)
+      if issue.present? && issue.status_id == 2
```

Comment:
@Quardah Can you please make the statuses (2, 14 etc) as configurable in a YML file, so that anyone can change it easily? 
Like, `if issue.present? && issue.status_id == CURRENT_REDMINE_STATE` and `issue.update(status_id: NEXT_REDMINE_STATE)` where these constants are either coming from YML or ENV variables.

Thanks 👍

Reply:
Hi. I just did patch as you asked. I struggled with the YAML so i went with ENV because i figured out because you already ask the user to setup ENV Github Token they would be familiar in doing so.

I've also added failsafes as defaults 2 and 3 for In Progress and In Review.

Please see issue #6 when you have a moment too.

Thank you.

Reply:
@Quardah Having a failsafe on a change of status for a project management platform seems a bit risky for me. It could break completely the ticket organization of a team before they understand that they needed to put an ENVVAR.

I think that it would be better to do nothing and fail without error. Or, to be perfect, to trigger an email to the admin email of Redmine.